lib/src/repositories/token_wallet_repository/token_wallet_repository_impl.dart

```diff
@@ -134,15 +164,15 @@ mixin TokenWalletRepositoryImpl implements TokenWalletRepository {
   void unsubscribeToken(Address owner, Address rootTokenContract) {
     final wallet = removeTokenWalletInst(owner, rootTokenContract);
     tokenPollingQueues.remove((owner, rootTokenContract))?.stopPolling();
-    wallet?.dispose();
+    wallet?.wallet?.dispose();
```

Comment:
Should we provide dispose() method in *WalletState classes that will dispose *Wallet?

Reply:
Is this really necessary? I'm not sure that state, that may contain error should know something about real wallet

Reply:
Hmmm. It seems you are right, because the wallet is created externally, and should be manages externally.